src/z3c/jbot/manager.py

```diff
@@ -87,14 +92,13 @@ def unregisterDirectory(self, directory):
 
         templates = []
 
-        for template, filename in self.templates.items():
+        for template, filename in list(self.templates.items()):
```

Comment:
This is not necessary: you're not adding/removing items to/from self.templates inside the loop.